src/App/EditPage.xaml.cs

```diff
@@ -155,15 +155,58 @@ private async Task<bool> On_BackRequested()
             return false;
         }
 
-        private void BgDeleteButton_Click(object sender, RoutedEventArgs e)
+        private void CancelDelete_Click(object sender, RoutedEventArgs e)
         {
-            BackgroundTasksCollection.Remove(GetTaskFromButton(sender as Button));
-            listEdited = true;
+            DeleteFlyout.Hide();
+        }
 
-            if (BackgroundTasksCollection.Count == 0)
+        /// <summary>
+        /// Confirm delete flyout
+        /// </summary>
+        private async void ConfirmDelete_Click(object sender, RoutedEventArgs e)
+        {
+            try
             {
-                BgTasksHeader.Visibility = Visibility.Collapsed;
+                if (activeTaskIsNowBg)
+                {
+                    BackgroundTasksCollection.Remove(activeTask);
+                    activeTaskIsNowBg = false;
+                }
+                else
+                {
+                    TasksCollection.Remove(activeTask);
+                }
+
+                listEdited = true;
+                if (TasksCollection.Count == 0)
+                {
+                    TasksHeader.Visibility = Visibility.Collapsed;
+                }
             }
+            catch (Exception ex)
+            {
+                if (ex.GetType() != typeof(FactoryOrchestratorConnectionException))
+                {
+                    ContentDialog failedDeleteDialog = new ContentDialog
+                    {
+                        Title = resourceLoader.GetString("FOXMLSaveFailed"),
+                        Content = $"{ex.Message}",
+                        CloseButtonText = resourceLoader.GetString("Ok")
+                    };
+
+                    _ = await failedDeleteDialog.ShowAsync();
+                }
+            }
+            DeleteFlyout.Hide();
+        }
+
+        private void BgDeleteButton_Click(object sender, RoutedEventArgs e)
+        {
+            var button = sender as Button;
+            activeTask = GetTaskFromButton(button);
+            activeTaskIndex = TasksCollection.IndexOf(activeTask);
+            activeTaskIsNowBg = true;

```

Comment:
nit these names should be activeTaskList* or activeList

Reply:
Could you elaborate? Wouldn't it be the tasks within the tasklist that would be edited/deleted here.

Reply:
doh, you are right! :)

Reply:
Aha! Thank you for the clarification.